lib/testers/rule-tester.js

```diff
@@ -396,10 +396,14 @@ class RuleTester {
          * @private
          */
         function assertASTDidntChange(beforeAST, afterAST) {
-            if (!lodash.isEqual(beforeAST, afterAST)) {
 
-                // Not using directly to avoid performance problem in node 6.1.0. See #6111
+            // Feature detect the Node.js implementation.
+            if (util.isDeepStrictEqual) {
                 assert.deepStrictEqual(beforeAST, afterAST, "Rule should not modify AST.");
+
+            // Do not use `assert.deepStrictEqual` in Node.js 6 and 7 due to performance issues, see #6111.
```

Comment:
Looks like `util.isDeepStrictEqual` was added in node 9, so it won't run in node 8 either, right?

Reply:
Uh, yes. Seems like it was not backported. I thought that would be the case.